srml/balances/src/lib.rs

```diff
@@ -399,11 +468,151 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
 	}
 }
 
+/// Opaque, move-only struct with private fields that serves as a token denoting that
+/// there are more funds in the system than is accounted for in total issuance.
+#[must_use]
+pub struct PositiveImbalance<T: Subtrait<I>, I: Instance=DefaultInstance>(T::Balance);
+
+/// Opaque, move-only struct with private fields that serves as a token denoting that
+/// there are more funds in the system than is accounted for in total issuance.
+#[must_use]
+pub struct NegativeImbalance<T: Subtrait<I>, I: Instance=DefaultInstance>(T::Balance);
+
+impl<T: Trait<I>, I: Instance> Imbalance<T::Balance> for PositiveImbalance<T, I> {
+	type Opposite = NegativeImbalance<T, I>;
+
+	fn zero() -> Self {
+		Self(Zero::zero())
+	}
+	fn drop_zero(self) -> result::Result<(), Self> {
+		if self.0.is_zero() {
+			Ok(())
+		} else {
+			Err(self)
+		}
+	}
+	fn split(self, amount: T::Balance) -> (Self, Self) {
+		let first = self.0.min(amount);
+		let second = self.0 - first;
+		(Self(first), Self(second))
+	}
+	fn merge(self, other: Self) -> Self {
+		Self(self.0 + other.0)
+	}
+	fn subsume(&mut self, other: Self) {
+		self.0 += other.0
```

Comment:
Why is this safe? The docs at the declaration of `subsume` doesn't mention anything about overflow/panics. 

The same question is about `merge`.

Reply:
It's not inherently safe, but imbalances are handled after everything else so if we got this far, it's unlikely that we'll overflow here. I'll switch to saturating anyway, but there's not much point panicking, which is the only other thing we can sensibly do.